pkg/testcase/clusterrestore.go

```diff
@@ -0,0 +1,416 @@
+			"node-external-ip: %s\n"+
+			"cluster-init: true\n"+
+			"EOF'", product, newClusterIP)
+	}
+
+	path := fmt.Sprintf("/etc/rancher/%s/", product)
+	cmd := fmt.Sprintf("sudo  mkdir -p %s && %s", path, createConfigFileCmd)
+
+	_, mkdirCmdErr := shared.RunCommandOnNode(cmd, newClusterIP)
+	Expect(mkdirCmdErr).NotTo(HaveOccurred())
+}
+
+func installProduct(
+	cluster *shared.Cluster,
+	newClusterIP string,
+	version string,
+) {
+	setConfigFile(cluster.Config.Product, newClusterIP)
+
+	installCmd := "curl -sfL "
+	if cluster.Config.Product == "k3s" {
+		installCmd += fmt.Sprintf("https://get.%s.io/ | sudo INSTALL_%s_VERSION=%s  INSTALL_%s_SKIP_ENABLE=true sh -",
+			cluster.Config.Product, strings.ToUpper(cluster.Config.Product), version,
+			strings.ToUpper(cluster.Config.Product))
+	} else {
+		installCmd += fmt.Sprintf("https://get.%s.io | sudo INSTALL_%s_VERSION=%s sh -",
+			cluster.Config.Product, strings.ToUpper(cluster.Config.Product), version)
+	}
+
+	_, installCmdErr := shared.RunCommandOnNode(installCmd, newClusterIP)
+	Expect(installCmdErr).NotTo(HaveOccurred())
+}
+
+func enableAndStartService(
+	cluster *shared.Cluster,
+	newClusterIP string,
+) {
+	_, enableServiceCmdErr := shared.ManageService(cluster.Config.Product, "enable", "server",
+		[]string{newClusterIP})
+	Expect(enableServiceCmdErr).NotTo(HaveOccurred())
+	_, startServiceCmdErr := shared.ManageService(cluster.Config.Product, "start", "server",
+		[]string{newClusterIP})
+
+	shared.LogLevel("info", "Starting service, waiting for service to complete background processes.")
+	Expect(startServiceCmdErr).NotTo(HaveOccurred())
+
+	time.Sleep(300 * time.Second)
+
+	statusServiceCmdRes, statusServiceCmdErr := shared.ManageService(cluster.Config.Product, "status", "server",
+		[]string{newClusterIP})
+	Expect(statusServiceCmdErr).NotTo(HaveOccurred())
+	fmt.Println("STATUS SERVICE OUT: ", statusServiceCmdRes)
+}
+
+func restoreS3Snapshot(
+	cluster *shared.Cluster,
+	onDemandPath,
+	token string,
+	newClusterIP string,
+	flags *customflag.FlagConfig,
+) {
+	var (
+		restoreCmdRes string
+		restoreCmdErr error
+	)
+
+	productLocationCmd, findErr := shared.FindPath(cluster.Config.Product, newClusterIP)
+	Expect(findErr).NotTo(HaveOccurred())
+
+	restoreCmd := fmt.Sprintf("sudo %s server --cluster-reset --etcd-s3 --cluster-reset-restore-path=%s"+
+		" --etcd-s3-bucket=%s --etcd-s3-folder=%s --etcd-s3-region=%s --etcd-s3-access-key=%s"+
+		" --etcd-s3-secret-key=%s --token=%s",
+		productLocationCmd,
+		onDemandPath,
+		flags.S3Flags.Bucket,
+		flags.S3Flags.Folder,
+		cluster.Aws.Region,
+		awsConfig.AccessKeyID,
+		awsConfig.SecretAccessKey,
+		token)
+
+	switch cluster.Config.Product {
+	case "k3s":
+		restoreCmdRes, restoreCmdErr = shared.RunCommandOnNode(restoreCmd, newClusterIP)
+		Expect(restoreCmdErr).NotTo(HaveOccurred())
+		Expect(restoreCmdRes).To(ContainSubstring("Managed etcd cluster"))
+		Expect(restoreCmdRes).To(ContainSubstring("has been reset"))
+	case "rke2":
```

Comment:
add comments here as well, why u validate the stdout from one product and other u validate the stderr